dask/array/core.py

```diff
@@ -1407,6 +1407,11 @@ def _blocks(self, index):
     def blocks(self):
         """ Slice an array by blocks
 
+        This allows blockwise slicing of a Dask array.  You can perform normal
+        Numpy-style slicing but now rather than slice elements of the array you
+        slice along blocks so, for example, ``x.blocks[0, ::2]`` produces a new
+        dask array with every other block in the first row of blocks.
+
```

Comment:
Maybe add:

> You can index blocks in any way that could index a numpy array of shape ``tuple(map(len, array.chunks))``. Integer indices ``k`` are converted internally into a slice object ``k:k+1``, to ensure that the array does not lose dimensions.

Reply:
Would simplify `tuple(map(len, array.chunks))` to `array.numblocks`.